tfhe/src/core_crypto/algorithms/slice_algorithms.rs

```diff
@@ -245,15 +250,93 @@ pub fn slice_wrapping_sub_scalar_mul_assign<Scalar>(
 ) where
     Scalar: UnsignedInteger,
 {
+    struct Impl<'a, Scalar> {
+        lhs: &'a mut [Scalar],
+        rhs: &'a [Scalar],
+        scalar: Scalar,
+    }
+
+    impl<Scalar: UnsignedInteger> pulp::NullaryFnOnce for Impl<'_, Scalar> {
+        type Output = ();
+
+        #[inline(always)]
+        fn call(self) -> Self::Output {
+            let Self { lhs, rhs, scalar } = self;
+
+            macro_rules! spec_constant {
+                ($constant: expr) => {
+                    if scalar == Scalar::cast_from($constant) {
```

Comment:
I think it would be cleaner to have
```
let signed_constant: i128 = $constant;
Scalar::cast_from(signed_constant as u128)
```
so that you can have simple signed constants without type annotations in the macro call

Reply:
just cast the constant directly in the Scalar::cast_from

Reply:
Defining `signed_constant` does not force us to add type annotations in the macro call

Reply:
I'm not a fan of the multiple let statements this creates 🤔 plus in rust most of the time signed constants are i32 so yeah I think I like the fact it is explicit here ?

Reply:
plus I put a comment above the spec constant chain